tests/StreamTest.cpp

```diff
@@ -87,6 +87,34 @@ static int s_StreamTestRead(struct aws_allocator *allocator, void *ctx)
 
 AWS_TEST_CASE(StreamTestRead, s_StreamTestRead)
 
+static int s_StreamTestReadEmpty(struct aws_allocator *allocator, void *ctx)
+{
+    (void)ctx;
+    {
+        Aws::Crt::ApiHandle apiHandle(allocator);
+
+        auto stringStream = Aws::Crt::MakeShared<Aws::Crt::StringStream>(allocator, "");
+
+        Aws::Crt::Io::StdIOStreamInputStream wrappedStream(stringStream, allocator);
+
+        aws_byte_buf buffer;
+        AWS_ZERO_STRUCT(buffer);
+        aws_byte_buf_init(&buffer, allocator, 256);
+
+        ASSERT_SUCCESS(aws_input_stream_read(wrappedStream.GetUnderlyingStream(), &buffer));
+
+        ASSERT_TRUE(buffer.len == 0);
+
+        aws_byte_buf_clean_up(&buffer);
+    }
+
+    Aws::Crt::TestCleanupAndWait();
+
+    return AWS_OP_SUCCESS;
+}
+
+AWS_TEST_CASE(StreamTestReadEmpty, s_StreamTestReadEmpty)
+
```

Comment:
This was somehow pulled in from merged commit: https://github.com/awslabs/aws-crt-cpp/commit/8e4da9be463cd74d1a9f9f7ff4f113f3b85c4c9a